catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/AsyncFileEntry.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import java.io.File;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+import javax.validation.constraints.NotNull;
+
+/**
+ * The core of this file is based upon {@link org.apache.commons.io.monitor.FileEntry}, modified to
+ * update only on successful async operations.
+ *
+ * <p>A wrapper class for a Java {@link File} object. This essentially keeps meta-snapshots taken
+ * when commit() is called. hasChanged will compare the current value of the file to the last time
+ * it's been committed, or the snapshot.
+ *
+ * @apiNote once hasChanged returns true, the user must commit the file once it's finished
+ *     processing to create a new meta-snapshot.
+ */
+public class AsyncFileEntry implements Serializable, Comparable<AsyncFileEntry> {
```

Comment:
✏️ Is `exists()` used anywhere? Can it be deleted?
✏️ Are `getLength()`, `hasChanged()`, and `hasChild(AsyncFileEntry)` used outside of this class? Can they be made `private`?

Reply:
hasChanged is used outside the class, for the rest though, they can be made private.